client/webserver/webserver.go

```diff
@@ -277,7 +283,10 @@ type WebServer struct {
 // be left blank, in which case a handful of default locations will be checked.
 // This will work in most cases.
 func New(cfg *Config) (*WebServer, error) {
-	log = cfg.Logger
+
+	if cfg.Logger != nil {
+		log = cfg.Logger
+	}
```

Comment:
I'm not actually sure this is the most elegant solution btw, suggestions appreciated.

Reply:
Can you just make a new logger for every test? Is the problem using the same logger for multiple tests at the same time? I thought the logger could handle concurrent logs though...

Reply:
I see the problem now, we change the logger as it is being used.

imo the correct way to solve this is to make `log` part of the `WebServer struct` and remove the package level one. It can't be used before being initiated with `New` anyhow, so there's no reason for it to be package level.